WordPress/gutenberg
# Add Typography text-orientation (Upright and mixed letters)

## What?

Adds a text-orientation typography block support, to be combined with the writing mode block support.
Adds support for text-orientation to paragraph blocks.

Follow up to https://github.com/WordPress/gutenberg/pull/50822
Closes https://github.com/WordPress/gutenberg/issues/50319

## Testing Instructions
This setting is disabled by default and can be enabled for themes that include theme.json.

1. In theme.json, under settings > typography, add `writingMode: true` and `textOrientation: true`.
2. Under styles > blocks, add
```
"core/paragraph": {
	"typography": {
		"writingMode": "vertical-rl",
		"textOrientation": "upright"
	}
}
```
3. Activate the theme. Confirm that all paragraphs are vertical and upright, top to bottom.
4. Remove the block style from theme.json.
5. Go to the Site Editor, Styles, Blocks, Paragraph, Typography, and set the writing mode to vertical and text-orientation to upright. Confirm that the styles are applied.
6. Change the writing mode to horizontal and confirm the change is applied.
7. Next, select a paragraph block and find the text orientation setting in the block settings sidebar, typography panel. It should not be displayed by default, but be available from the three dot menu.
8. Confirm that all four writing mode and text orientation options work in the editor and front.
9. Change the site language to use an RTL language, and repeat step 7. 

## Screenshots or screencast 

https://github.com/WordPress/gutenberg/assets/7422055/d579f804-f90f-4fbe-8b94-e044692ae4e4
